docs/tutorials/typescript/subscriptions/index.md

```diff
@@ -1,416 +1,417 @@
 tags: [TypeScript, SDK]
 last_update:
-  date: 2021-10-01
+  date: 2024-06-07
 title: Build a subscription workflow with Temporal and TypeScript
-description: In this tutorial, we will tour all of the Workflow APIs you should know, primarily Signals, Queries, `condition`, and `sleep`, by building a realistic monthly subscription payments workflow that can be canceled and changed while it runs.
+description: Implement a subscription application using Temporal's Workflows, Activities, Signals, and Queries, enabling the payment workflow to be canceled or modified during execution.
 image: /img/temporal-logo-twitter-card.png
 ---
 
 ![Temporal TypeScript SDK](/img/sdk_banners/banner_typescript.png)
 
-import { OutdatedNotice } from '@site/src/components'
-
-<OutdatedNotice />
-
 ## Introduction
 
-In this tutorial, you will tour all of the [Workflow APIs](https://docs.temporal.io/typescript/workflows#workflow-apis) you should know, primarily Signals, Queries, `condition`, and `sleep` (and eventually Child Workflows and `continueAsNew`), by building a realistic monthly subscription payments workflow that can be canceled and changed while it runs.
-
-- The goal is to give you a more accessible introduction to these APIs by explaining them in context of a realistic application.
-- We also give an example of how you break down project requirements into Activities and Workflow logic.
-
-Your task is to write a Workflow for a limited time Subscription (eg a 36 month Phone plan) that satisfies these conditions:
-
-1. When the user signs up, **send a welcome email** and start a free trial for `TrialPeriod`.
-2. When the `TrialPeriod` expires, start the billing process
-   - If the user cancels during the trial, **send a trial cancellation email**.
-3. Billing Process:
-   - As long as you have not exceeded `MaxBillingPeriods`,
-   - **Charge the customer** for the `BillingPeriodChargeAmount`.
-   - Then wait for the next `BillingPeriod`.
-   - If the customer cancels during a billing period, **send a subscription cancellation email**.
-   - If Subscription has ended normally (exceeded `MaxBillingPeriods` without cancellation), **send a subscription ended email**.
-4. At any point while subscriptions are ongoing, be able to look up and change any customer's:
-   - Amount Charged
-   - Period number (for manual adjustments e.g. refunds)
-
-Of course, this all has to be fault tolerant, scalable to millions of customers, testable, maintainable, and observable.
+In this tutorial, you'll build a subscription application using Temporal and TypeScript. You'll use Temporal Workflows, Activities, sleep, Signals, and Queries to manage the core subscription logic. By leveraging Temporal's capabilities, you can handle complex subscription scenarios directly within Temporal's environment, reducing the need for external databases or queues. This reduces the complexity of the code you have to write and maintain.
```

Comment:
> This reduces the complexity of the code you have to write and maintain.

Quite the claim....

What would it traditionally look like using a Database?

Also, unless they are using Temporal Cloud there is a giant Elephant in the room that the code complexity is reduced at the expense of operating a Temporal Service... I think we need to address that.

Also that at a small scale - Temporal might be overkill. One major value add here is that if the subscription application is used by thousands and thousands of folks and there is a need to have many servers available to handle the load - then it can scale out indefinitely.

I think this introduction glosses over these value props - 

I would argue that in a small scale environment - it *could*  be easier to hook up to a database and use basic read / write operations to achieve the same functionality.
So if we want a strong introduction, then I would say we need to have a strong comparison to traditional and more explicit call outs to the value props that Temporal brings.

Are there any other value propositions here?
What Workflow pattern is being expressed here? (hint: Workflow Entity)

Reply:
This is a great callout - thanks!